stp_core/common/logging/TimeAndSizeRotatingFileHandler.py

```diff
@@ -8,91 +13,122 @@ class TimeAndSizeRotatingFileHandler(TimedRotatingFileHandler, RotatingFileHandl
 
     def __init__(self, filename, when='h', interval=1, backupCount=0,
                  encoding=None, delay=False, utc=False, atTime=None,
-                 maxBytes=0, compress=False):
+                 maxBytes=0, compression=None):
 
         TimedRotatingFileHandler.__init__(self, filename, when, interval,
                                           backupCount, encoding, delay,
                                           utc, atTime)
         self.maxBytes = maxBytes
-        self.compress = compress
+        self.compression = compression
+        self.compressor = None
+
+        log_prefix = os.path.basename(self.baseFilename)
+        self.log_pattern = re.compile("^{}(?:|\.(\d+)(?:|\.gz|\.xz))$".format(log_prefix))
```

Comment:
Do we have a test for the new log file patten and that the date is correctly excluded?

Reply:
@ashcherbakov no, we don't have those tests. Do we need just simple tests for this regex, or something more complex like that all log files created match this regex, or maybe both? 

As for date exclusion - in fact file naming was rewritten from scratch so we are in full control of it now, so I'm not sure that we need explicit test that there's no date in filename, especially if we implement above mentioned tests.

Reply:
It would be great to have both

Reply:
@ashcherbakov done